nion/swift/ImageCanvasItem.py

```diff
@@ -255,34 +260,86 @@ def _repaint(self, drawing_context: DrawingContext.DrawingContext, canvas_bounds
         scale_marker_height = 6
         scale_marker_font = "normal 14px serif"
         get_font_metrics_fn = self.__get_font_metrics_fn
-        if dimensional_calibration:  # display scale marker?
+
+        if dimensional_calibration and screen_pixel_per_image_pixel and screen_pixel_per_image_pixel > 0.0:
+            scale_marker_image_width = scale_marker_width / screen_pixel_per_image_pixel
+            calibrated_scale_marker_width = Geometry.make_pretty2(scale_marker_image_width * dimensional_calibration.scale, True)
+            scale_marker_image_width = calibrated_scale_marker_width / dimensional_calibration.scale
+            scale_marker_calculated_width = scale_marker_image_width * screen_pixel_per_image_pixel
+
+
+            text1 = dimensional_calibration.convert_to_calibrated_size_str(scale_marker_image_width)
+            fm1 = get_font_metrics_fn(scale_marker_font, text1)
+            text_height = float(fm1.height-5)
+            if self.__include_info_text:
+                text2 = info_text
+                fm2 = get_font_metrics_fn(scale_marker_font, text2)
+                text_width = max(fm1.width, fm2.width)
+            else:
+                text2 = ""
+                text_height = text_height/2
+                text_width = fm1.width
+            padding = 3
+
+            # Determine width needed to fit both bar and text
+            content_width = max(scale_marker_calculated_width, text_width)
+
+            # Set drawing origin
+            if self.__draw_on_right:
+                origin_x = max(0.0, canvas_bounds.width * 2 - (content_width + padding * 2 + 16))
+            else:
+                origin_x = 0
+
             with drawing_context.saver():
-                drawing_context.translate(canvas_bounds.left, canvas_bounds.top)
-                if screen_pixel_per_image_pixel and screen_pixel_per_image_pixel > 0.0:
-                    scale_marker_image_width = scale_marker_width / screen_pixel_per_image_pixel
-                    calibrated_scale_marker_width = Geometry.make_pretty2(scale_marker_image_width * dimensional_calibration.scale, True)
-                    # update the scale marker width
-                    scale_marker_image_width = calibrated_scale_marker_width / dimensional_calibration.scale
-                    scale_marker_calculated_width = scale_marker_image_width * screen_pixel_per_image_pixel
-                    baseline = canvas_bounds.height
+                drawing_context.translate(canvas_bounds.left + origin_x, canvas_bounds.top)
+                baseline = canvas_bounds.height
+
+                # Background box
+                if self.__background:
+                    total_text_height = text_height * 2 + padding
+                    total_height = scale_marker_height + padding + total_text_height
```

Comment:
I suggest removing the option to not draw the info text, at least in this PR. But if it were to be kept, the height calculation needs to take into account when it is empty.

<img width="224" alt="image" src="https://github.com/user-attachments/assets/18a2127e-bb9f-4e1e-9c9e-97daf404780a" />

Reply:
I don't think a checkbox for enabling this is the right way to go. Instead, I think we should allow the user to decide what information they put into the info line, if any. But until we settle on that UI, can we just remove it from this PR?